lib/cdo/contact_rollups.rb

```diff
@@ -378,36 +380,36 @@ def self.update_geo_data_from_forms(form_kind)
       email = data['email_s'].presence
       next if email.nil?
 
-      # get user-supplied address/location data from form if present, from any of the differently-named location fields across all form kinds
+      # Get user-supplied address/location data from form if present, from any of the differently-named location fields across all form kinds
       street_address = data['location_street_address_s'].presence || data['user_street_address_s'].presence
       city = data['location_city_s'].presence || data['user_city_s'].presence || data['teacher_city_s'].presence
       state = data['location_state_s'].presence || data['teacher_state_s'].presence
       if state.nil?
-        # note that the 'user_state_s' record is in fact a *state code*, not a state name
+        # Note that the 'user_state_s' record is in fact a *state code*, not a state name
         state_code = data['location_state_code_s'].presence || data['state_code_s'].presence || data['user_state_s'].presence
         state = get_us_state_from_abbr(abbr: state_code, include_dc: true).presence unless state_code.nil?
       end
       postal_code = data['location_postal_code_s'].presence || data['zip_code_s'].presence || data['user_postal_code_s'].presence
       country = data['location_country_s'].presence || data['country_s'].presence || data['teacher_country_s'].presence
 
-      # skip if this form data has no address/location info at all
+      # In practice, self-reported country data (often free text) from forms is garbage. The exception is a frequent reliable
+      # value of exactly "united states", which is what gets filled in automatically if you enter a zip code on the petition form.
+      # Trust this value if we see it, otherwise do not use country from forms; fall back to any country we got from IP geo lookup.
+      country = nil unless country.presence && country.downcase == UNITED_STATES
```

Comment:
Though unclear if a readability gain (see recent `Slack#dashboard-dev`), this could be shortened to
`country = nil unless country.try(:downcase) == UNITED_STATES`.